sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/Analyzer.scala

```diff
@@ -2529,19 +2529,19 @@ class Analyzer(override val catalogManager: CatalogManager) extends RuleExecutor
      */
     private def resolveSubQueries(plan: LogicalPlan, outer: LogicalPlan): LogicalPlan = {
       plan.transformAllExpressionsWithPruning(_.containsPattern(PLAN_EXPRESSION), ruleId) {
-        case s @ ScalarSubquery(sub, _, exprId, _, _, _) if !sub.resolved =>
+        case s @ ScalarSubquery(sub, _, exprId, _, _, _) if !sub.analyzed =>
           resolveSubQuery(s, outer)(ScalarSubquery(_, _, exprId))
```

Comment:
can we make sure we set the `analyzed` flag in `resolveSubQuery`?

Reply:
I added `checkAnalysis` in `resolveSubQuery`, is it ok?

Reply:
> I added `checkAnalysis` in `resolveSubQuery`, is it ok?

It doesn't seem to work, the test cases failed.